spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/event/Event.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi.event;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/**
+ * The Event base class.
+ * It provides a default field "at" that gives the timestamp when the event was created.
+ * When serialized to JSON, will add a "type" field with the name of the runtime class name.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", include = JsonTypeInfo.As.EXTERNAL_PROPERTY)
+public abstract class Event {
```

Comment:
Do you think it would be worthwhile to have the `at` field initialized with the current time by default? That would reduce the amount of duplicated code greatly, because then event sources wouldn't have to call `at(clock.millis())`.  
The general use of `Clock` makes this more cumbersome, but we could create a factory method of sorts, just to make it a bit sleeker to use, wdyt?

Reply:
@paullatzelsperger this could go in the discussion about the event envelope mentioned in a comment above, if we do that we could create the event instance separately and then having someone (also the event router`) that would wrap the event with an envelope with other informations like `at`, an id, the type, and so on.

Reply:
> @jimmarino @paullatzelsperger , I added a commit where I introduced the `EventEnvelope` class, that contains the event and the metadata about the event (for the moment only the timestamp, but other information could be included in the future, as an id, the connector id and so on). I'm in doubt about the naming, because I'm not sure that the `Event` class now represent an event, since it misses the timestamp, an idea could be to rename `EventEnvelope` to `Event` and `Event` to `EventData` or something like this. wdyt?

Agreed the "Event" class should be the top-level name. Why not collapse the two classes into a single Event class? Generally, envelopes are introduced so that code can either process parts of a message without having to access the body (e.g. encrypted data) and/or to "unwrap" a message so that header information does not pollute non-transport areas of a code base.

BTW, this is a great PR! :-)

Reply:
@jimmarino so your advice would be to have everything (data and metadata) inside the `Event` class as it was before my last commit?